blocks/library/text/index.js

```diff
@@ -29,13 +29,15 @@ registerBlockType( 'core/text', {
 		content: query( 'p', children() ),
 	},
 
+	pasteMatcher: ( node ) => node.nodeName === 'P',
```

Comment:
Could we find a more "generic" name, because this could be used to convert legacy posts too.